tests/formatter_tests.js

```diff
@@ -80,6 +83,134 @@ VF.Test.Formatter = (function() {
+      var score = vf.EasyScore();
+
+      var notes11 = score.notes('a4/2, a4/4, a4/8, ab4/16, an4/16');
+      var voice11 = score.voice(notes11, { time: '4/4' });
+
+      var notes21 = score.notes('c4/2, d4/8, d4/8, e4/8, e4/8');
+      var voice21 = score.voice(notes21, { time: '4/4' });
+
+      var beams = VF.Beam.generateBeams(notes11.slice(2));
+      beams = beams.concat(beams, VF.Beam.generateBeams(notes21.slice(1, 3)));
+      beams = beams.concat(VF.Beam.generateBeams(notes21.slice(3)));
+      var formatter = vf.Formatter(options.params)
+        .joinVoices([voice11])
+        .joinVoices([voice21]);
+
+      var width = formatter.preCalculateMinTotalWidth([voice11, voice21]);
+      var stave11 = vf.Stave({ y: 20, width: width + 20 });
+      var stave21 = vf.Stave({ y: 130, width: width + 20 });
+      formatter.format([voice11, voice21], width);
+      vf.StaveConnector({
+        top_stave: stave11,
+        bottom_stave: stave21,
+        type: 'brace',
+      });
+      var ctx = vf.getContext();
+      stave11.setContext(ctx).draw();
+      stave21.setContext(ctx).draw();
+      voice11.draw(ctx, stave11);
+      voice21.draw(ctx, stave21);
+      beams.forEach(function(b) {
+        b.setContext(ctx).draw();
+      });
+      ok(true);
+    },
+    unalignedNoteDurations: function(options) {
+      var vf = VF.Test.makeFactory(options, 600, 400);
+      var score = vf.EasyScore();
+
+      var notes11 = [
+        new VF.StaveNote({ keys: ['a/4'], duration: '8' }),
+        new VF.StaveNote({ keys: ['b/4'], duration: '4' }),
+        new VF.StaveNote({ keys: ['b/4'], duration: '8' })
+      ];
+      var notes21 = [
+        new VF.StaveNote({ keys: ['a/4'], duration: '16' }),
+        new VF.StaveNote({ keys: ['b/4.'], duration: '4' }),
+        new VF.StaveNote({ keys: ['a/4'], duration: '8d' }).addDotToAll()
+      ];
+
+      var ctx = vf.getContext();
+      var voice11 = score.voice(notes11, { time: '2/4' }).setMode(VF.Voice.Mode.SOFT);
+      var voice21 = score.voice(notes21, { time: '2/4' }).setMode(VF.Voice.Mode.SOFT);
+      var beams21 = VF.Beam.generateBeams(notes21);
+      var beams11 = VF.Beam.generateBeams(notes11);
+      var formatter = new VF.Formatter();
+      formatter.joinVoices([voice11]);
+      formatter.joinVoices([voice21]);
+      var width = formatter.preCalculateMinTotalWidth([voice11, voice21]);
+      var stave11 = vf.Stave({ y: 20, width: width + 20 });
+      var stave21 = vf.Stave({ y: 130, width: width + 20 });
+      formatter.format([voice11, voice21], width);
+      stave11.setContext(ctx).draw();
+      stave21.setContext(ctx).draw();
+      voice11.draw(ctx, stave11);
+      voice21.draw(ctx, stave21);
+      beams21.forEach(function(b) {
+        b.setContext(ctx).draw();
+      });
+      beams11.forEach(function(b) {
+        b.setContext(ctx).draw();
+      });
+      if (voice11.tickables[1].getX() >= voice21.tickables[1].getX()) {
```

Comment:
@AaronDavidNewman the second note of voice11 should be to the right of second note of voice21. In the if condition checking the failure, it should be, I believe, <= and not >=.
```js
if (voice11.tickables[1].getX() <= voice21.tickables[1].getX())
```
@0xfe should the alignment between staves consider the value getWidth of the ticketables? I mean should the condition for the warning rather be:
```js
if (voice11.tickables[1].getX() < (voice21.tickables[1].getX() + voice21.tickables[1].getWidth()))
```

Reply:
No need to consider `getWidth()` if formatting is already done.